### PR TITLE
Sample load acceptance test

### PR DESCRIPTION
**Pull request by liamtoozer**
# Motivation and Context

The main purpose of this change is to allow future tests to be run independently of each other. 

# What has changed

* Unique string generation for independent tests
* Remove db teardown (keep teardown SQL for local dev)
* Refactor of existing tests
* Move steps to [domain concepts](https://docs.cucumber.io/guides/anti-patterns/#how-to-decouple-steps-step-definitions)
*  Confirms sample load success by sample unit ID retrieval from casesvc

# How to test?

**Note**: Currently only able to test locally
1. Checkout the branch from [this PR](https://github.com/ONSdigital/census-rm-docker-dev/pull/1) and start containers with `make up`.
2. `make test`

# Links
[Trello](https://trello.com/c/8Hq6ZFLV/494-sample-load-acceptance-test-8)

### CODE REVIEW

**Comment by liamtoozer** [comment]
[Sample load acceptance test (8)](https://trello.com/c/8Hq6ZFLV/494-sample-load-acceptance-test-8)

File: acceptance_tests/features/load_samples_to_db.feature [hunk 1]
```diff
@@ -1,10 +1,7 @@
...
 
-  Background: Clear DB
-    Given the database is cleared down
-
-  Scenario: Good sample file load GSFL
-    Given there is a live collection exercise with unique id "GSFL"
+  Scenario: Successful sample file upload
+    Given a survey exists with collection exercise
     When a sample file "Sample_10.csv" is loaded
     Then a call to the casesvc api returns 10 cases
```
Comment (AdamHawtin) [inline acceptance_tests/features/load_samples_to_db.feature]: This 'Then' describes how its testing for the goal, not the actual goal of the test which is that the cases are created and persisted in the case service.

File: utilities/date_utilities.py [hunk 2]
```diff
@@ -1,44 +1,9 @@
 from datetime import datetime
-import maya
```
Comment (AdamHawtin) [inline utilities/date_utilities.py:2]: Can the dependency for `maya` be removed now?

**Review by AdamHawtin — CHANGES_REQUESTED** [review]
The test runs fine against local docker images, a few comments though.

Also the README needs updating.

File: README.md [hunk 3]
```diff
@@ -1,21 +1,10 @@
...
-
-### Build the Docker image
-
-> docker build -t rjweeks/census-rm-acceptance-tests .
-
-### Push the Docker image
-
-> docker push rjweeks/census-rm-acceptance-tests
-
+1. Clone `census-rm-docker-dev ` and run `make up` to start the required services
```
Comment (AdamHawtin) [inline README.md]: Could be a link to the repo?

File: resources/database/database_reset_rm.sql [hunk 4]
```diff
@@ -46,17 +46,6 @@ ALTER SEQUENCE action.casepkseq RESTART WITH 1;
...
 
-/* Clean Action Exporter DB */
-
-TRUNCATE actionexporter.actionrequest CASCADE;
-TRUNCATE actionexporter.address CASCADE;
-TRUNCATE actionexporter.contact CASCADE;
-TRUNCATE actionexporter.filerowcount CASCADE;
-
-ALTER SEQUENCE actionexporter.actionrequestpkseq RESTART WITH 1;
-ALTER SEQUENCE actionexporter.contactpkseq RESTART WITH 1;
```
Comment (AdamHawtin) [inline resources/database/database_reset_rm.sql]: Why has this been deleted?

Comment (AdeHarrison) [reply resources/database/database_reset_rm.sql]: Reinstated statements deleted in error

File: utilities/sample_loader/sample_file_loader.py [hunk 5]
```diff
@@ -36,7 +35,7 @@
 def load_sample_file(context):
```
Comment (LukeLoze1) [inline utilities/sample_loader/sample_file_loader.py:36]: stick RABBITMQ_SERVICE_HOST and other env variables in the config file along with these values

**Review by LukeLoze1 — CHANGES_REQUESTED** [review]

**Review by AdamHawtin — APPROVED** [review]

**Review by AdamHawtin — APPROVED** [review]

**Review by LukeLoze1 — APPROVED** [review]

**Pull request merged**